### DISCUSSION TITLE
Common extensions windows support?

### DISCUSSION THREAD

**Discussion by aolko** [crazywhalecc/static-php-cli#389] [Q&A] [answered]
When all the common extensions will be supported by windows `spc`? Namely:
```
apcu,bcmath,calendar,ctype,curl,dba,dom,exif,fileinfo,filter,gd,iconv,intl,mbregex,mbstring,mysqli,mysqlnd,opcache,openssl,pcntl,pdo,pdo_mysql,pdo_pgsql,pdo_sqlite,pdo_sqlsrv,pgsql,phar,posix,rar,readline,redis,session,simplexml,sockets,sodium,sqlite3,sqlsrv,tokenizer,uuid,xml,xmlreader,xmlwriter,xsl,yaml,zip,zlib,zstd
```

**Comment by crazywhalecc** [comment, marked as answer]
Some exetensions are supported in official distributions, but they all compiled dynamically (dll). Supporting all extensions is impossible, but we can draft a list.

It would be better if you could sort out the extensions you need one by one and which ones need to be supported as soon as possible (so I can have an expectation).

**Reply by aolko** [reply]
I need the default LAMP php extensions + apcu (cache) + intl (i18n) + sqlserver (pdo/sqlserv) + yaml + rar

**Reply by crazywhalecc** [reply]
Now we support apcu, sqlsrv, pdo_sqlsrv, yaml and rar: #488 , will merge soon. 

Intl extension cannot be statically compiled due to `icu` lib does not support building static libs. But you can use dynamic linked `intl.dll` with static php: `extension=path\to\intl.dll`.

**Reply by aolko** [reply]
> Now we support apcu, sqlsrv, pdo_sqlsrv, yaml and rar: #488 , will merge soon.
> 
> Intl extension cannot be statically compiled due to `icu` lib does not support building static libs. But you can use dynamic linked `intl.dll` with static php: `extension=path\to\intl.dll`.

can you elaborate on the 

`extension=path\to\intl.dll`

part?

**Reply by crazywhalecc** [reply]
Theoretically you just need to match `PHP Major Version`, `Visual Studio Major Version` and `(Non-)Thread Safety`. See: https://github.com/crazywhalecc/static-php-cli/pull/445 .

But I haven't test a lot, sometimes directly linking to dynamic extensions is not working due to missing other dlls.

**Comment by koossaayy** [comment]
How about the ZIP extension on Windows ?

**Reply by crazywhalecc** [reply]
😵‍💫 Will, but not fast recently.

**Reply by koossaayy** [reply]
It's okay, I understand that can take sometime. 
Thank you ♥.

**Reply by crazywhalecc** [reply]
Supported via #488 .
